tap_mavenlink/streams/base.py

```diff
@@ -13,14 +13,21 @@
 
 from tap_framework.streams import BaseStream as base
 
+from tap_mavenlink.utils import get_stream_metadata
 
 LOGGER = singer.get_logger()
 
-
 class BaseStream(base):
     KEY_PROPERTIES = ['id']
     CACHE = False
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.stream_metadata = get_stream_metadata(self.catalog.to_dict())
+        self.replication_method = self.stream_metadata.get('replication-method')
+        self.replication_key = self.stream_metadata.get('replication-key')
+
```

Comment:
@adrien-mg comment from first (bad) attempt at this PR
> Using this syntax I think it's possible that some streams end up with: self.replication_method == None and self.replication_key == None if those fields are not defined explicitly in the catalog's stream's metadata?
> Not sure if this is a problem at all though.

You're right. This shouldn't be a problem though, since the args validation at the top of the app will only allow sensible method-key combinations. That's the plan anyway 🤞

Reply:
:+1: